account_fiscal_year_closing/models/account_fiscalyear_closing.py

```diff
@@ -0,0 +1,701 @@
+                record.closing_template_id.id != record.stored_template_id.id
+            )
+
+    def _prepare_mapping(self, tmpl_mapping):
+        self.ensure_one()
+        dest_account = False
+        # Find the destination account
+        name = tmpl_mapping.name
+        if tmpl_mapping.dest_account:
+            dest_account = self.env["account.account"].search(
+                [
+                    ("company_id", "=", self.company_id.id),
+                    ("code", "=ilike", tmpl_mapping.dest_account),
+                ],
+                limit=1,
+            )
+            # Use an error name if no destination account found
+            if not dest_account:
+                name = _("No destination account '%s' found.") % (
+                    tmpl_mapping.dest_account,
+                )
+        return {
+            "name": name,
+            "src_accounts": tmpl_mapping.src_accounts,
+            "dest_account_id": dest_account,
+        }
+
+    @api.model
+    def _prepare_type(self, tmpl_type):
+        return {
+            "account_type_id": tmpl_type.account_type_id,
+            "closing_type": tmpl_type.closing_type,
+        }
+
+    def _get_default_journal(self, company):
+        """To be inherited if we want to change the default journal."""
+        journal_obj = self.env["account.journal"]
+        domain = [("company_id", "=", company.id)]
+        journal = journal_obj.search(
+            domain + [("code", "=", "MISC")],
+            limit=1,
+        )
+        if not journal:
+            journal = journal_obj.search(
+                domain + [("type", "=", "general")],
+                limit=1,
+            )
+        return journal
+
+    def _prepare_config(self, tmpl_config):
+        self.ensure_one()
+        mappings = self.env["account.fiscalyear.closing.mapping"]
+        for m in tmpl_config.mapping_ids:
+            mappings += mappings.new(self._prepare_mapping(m))
+        types = self.env["account.fiscalyear.closing.type"]
+        for t in tmpl_config.closing_type_ids:
+            types += types.new(self._prepare_type(t))
+        if tmpl_config.move_date == "last_ending":
+            date = self.date_end
+        else:
+            date = self.date_opening
+        return {
+            "enabled": True,
+            "name": tmpl_config.name,
+            "sequence": tmpl_config.sequence,
+            "code": tmpl_config.code,
+            "inverse": tmpl_config.inverse,
+            "move_type": tmpl_config.move_type,
+            "date": date,
+            "journal_id": (
+                tmpl_config.journal_id or self._get_default_journal(self.company_id).id
+            ),
+            "mapping_ids": mappings,
+            "closing_type_ids": types,
+            "closing_type_default": tmpl_config.closing_type_default,
+        }
+
+    # @api.onchange('closing_template_id')
+    # Not working due to https://github.com/odoo/odoo/issues/20163
```

Comment:
Is this still valid?

Reply:
Apparently not: https://github.com/odoo/odoo/issues/20163#issuecomment-863133625.
I enabled the onchange and removed both the method `action_load_template` and `is_new_template` that existed only to perform the onchange's job.

Tried the loading on onchange in my local environment and seems to be working fine